site/content/blog/2019-10-21-localizing-matterpoll/index.md

```diff
@@ -0,0 +1,188 @@
+
+Let's examine the difference between the two versions. In v1, translation strings are defined in the translation files; for example, `en-US.all.json` would contain:
+```json
+  {
+    "id": "settings_title",
+    "translation": "Settings"
+  }
+```
+These translations would then be accessible in the Go code with a translation function (`T("settings_title")`). This leads to a loose coupling between the translations string in the translation file and in the code. Unfortunately developers tend to forget to add a translation string or misspell the id.
+
+V2 fixed this by defining the translations string in the code and automatically extracting it into the translation file. This greatly improves the developer experience. In addition, there are now **two** translation files for every language. For example, German has `active.de.json`, which contains already translated strings, and `translate.de.json`, which contains strings that have to be translated. They are either newly-added strings or strings where the text changed. `goi18n` helps with auto-populating these files.
+
+This change is very handy for both developers and contributors and the main reason we ended up choosing v2 for Matterpoll. Other notable changes are:
+
+- Better support for plural forms with options for e.g., `Zero`, `One`, `Few` and `Many`.
+- No global state, which helps with running tests in parallel.
+- Use of [`golang.org/x/text/language`](https://godoc.org/golang.org/x/text/language) for standardized behavior.
+
+For further information refer to [Changelog](https://github.com/nicksnyder/go-i18n/blob/master/CHANGELOG.md#v2) of go-i18n.
+
+Also, it's a good idea to stay up to date with the major version of the libraries that you are using. This ensures you get the latest bug fixes because open libraries don't backport fixes.
+
+These improvements are why we decided to go with v2 of go-i18n.
+
+
+#### Choosing a File Format
+
+Go-i18n supports multiple file formats for the translation file. In fact, you can use any file format as long as you implement an [unmarshal function](https://godoc.org/github.com/nicksnyder/go-i18n/v2/i18n#UnmarshalFunc) yourself. The most common formats are `JSON`, `TOML` and `YAML`.
+
+What is the preferred format for us? `goi18n` uses `TOML` by default, but since the Mattermost server uses `JSON` we decided to stay with that format. We could just as easily have stuck with the default format 'TOML' of 'goi18n' for consistency with the library default with no adverse effect.
```

Comment:
```suggestion
What is the preferred format for us? `goi18n` uses `TOML` by default, but since the Mattermost server uses `JSON` we decided to stay with that format. We could just as easily have stuck with the default format `TOML` of 'goi18n' for consistency with the library default with no adverse effect.
```